storage/client/queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -0,0 +1,339 @@
+        try {
+            URL fullURL = new URL(endpoint);
+            this.endpoint = new URL(fullURL.getProtocol() + "://" + fullURL.getHost());
+
+            // Attempt to get the queue name from the URL passed
+            String[] pathSegments = fullURL.getPath().split("/", 2);
+            if (pathSegments.length == 2 && !ImplUtils.isNullOrEmpty(pathSegments[1])) {
+                this.queueName = pathSegments[1];
+            }
+
+            // Attempt to get the SAS token from the URL passed
+            SASTokenCredential credential = SASTokenCredential.fromQuery(fullURL.getQuery());
+            if (credential != null) {
+                this.sasTokenCredential = credential;
+            }
+        } catch (MalformedURLException ex) {
+            throw new IllegalArgumentException("The Azure Storage Queue endpoint url is malformed.");
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets the name of the queue that the client will interact with.
+     *
+     * @param queueName Name of the queue
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code queueName} is {@code null}.
+     */
+    public QueueClientBuilder queueName(String queueName) {
+        this.queueName = Objects.requireNonNull(queueName);
+        return this;
+    }
+
+    /**
+     * Sets the {@link SASTokenCredential} used to authenticate requests sent to the Queue service.
+     *
+     * @param credential SAS token credential generated from the Storage account that authorizes requests
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public QueueClientBuilder credential(SASTokenCredential credential) {
+        this.sasTokenCredential = Objects.requireNonNull(credential);
+        return this;
+    }
+
+    /**
+     * Creates a {@link SharedKeyCredential} from the {@code connectionString} used to authenticate requests sent to the
+     * Queue service.
+     *
+     * @param connectionString Connection string from the Access Keys section in the Storage account
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code connectionString} is {@code null}.
+     */
+    public QueueClientBuilder connectionString(String connectionString) {
+        Objects.requireNonNull(connectionString);
+        this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        Objects.requireNonNull(connectionString);
+        this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
+        return this;
+    }
+
+    private void getEndPointFromConnectionString(String connectionString) {
+        HashMap<String, String> connectionStringPieces = new HashMap<>();
+        for (String connectionStringPiece : connectionString.split(";")) {
```

Comment:
Should look into having a ConnectionStringParser class in Azure Core, I've been seeing this functionality in a lot of places.